lgc/state/PipelineState.cpp

```diff
@@ -1121,7 +1121,28 @@ unsigned PipelineState::getShaderWaveSize(ShaderStage stage) {
   assert(stage <= ShaderStageCompute);
   if (!m_waveSize[stage])
     setShaderDefaultWaveSize(stage);
-  return m_waveSize[stage];
+
+  unsigned waveSize = m_waveSize[stage];
+
+  if (stage == ShaderStageVertex) {
```

Comment:
Please check gfxip. We should have such handling for gfx9+.
Place this comment to explain the handling
// NOTE: For GFX9+, two shaders are merged as a shader pair. The wave size is determined by the larger one. That is
// to say:
//   - VS + TCS -> HW HS
//   - VS + GS -> HW GS (no tessellation)
//   - TES + GS -> HW GS
//   - VS/TES -> HW GS (NGG, no geometry)